### ISSUE TITLE
Catch spacing if called without argument

### ISSUE THREAD

**Issue by crystalfp** [eslint/eslint#13977]
The current implementation of `keyword-spacing` is not able to cover the two cases of `catch` clause with and without argument.

**What rule do you want to change?**
keyword-spacing

**Does this change cause the rule to produce more or fewer warnings?**
Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
Add in overrides an option `afterIfParen`. It works like the current `after` if the next character is a paren and not a brace.

**Please provide some example code that this change will affect:**
With the setting:
```js
/* eslint keyword-spacing: ["error", {"overrides": {"catch":{"after": false}}}] */
```
The following is correct:
```js
try {
  foo();
} catch(error) {
  bar();
}
```
But the following is marked incorrect:
```js
try {
  foo();
} catch {
  bar();
}
```

**What does the rule currently do for this code?**
It pass the first one and flags the second one. If the overrides is changed to: `{"catch":{"after": true}}` it pass the second one and flags the first.

**What will the rule do after it's changed?**
Both examples becomes valid.

**Are you willing to submit a pull request to implement this change?**
Unfortunately no.

**Comment by anikethsaha** [comment]
THanks for the issue.

I think [space-before-blocks](https://eslint.org/docs/rules/space-before-blocks#require-or-disallow-space-before-blocks-space-before-blocks) might be helpful here.

**Comment by crystalfp** [comment]
Unfortunately no.
I already have `space-before-blocks: warn` but I should have to put the eslint disable comment inline to disable the keyword-spacing warning on the catch line:
```
try {
  something();
}           
catch { // eslint-disable-line @typescript-eslint/keyword-spacing
  other();
}
```
It is typescript, but has the same problem shown by plain keyword-spacing rule.
Anyway, thanks for looking at it
mario
